subsys/net/ip/tcp.c

```diff
@@ -4051,7 +4065,7 @@ int net_tcp_finalize(struct net_pkt *pkt, bool force_chksum)
 
 	tcp_hdr->chksum = 0U;
 
-	if (net_if_need_calc_tx_checksum(net_pkt_iface(pkt)) || force_chksum) {
+	if (net_if_need_calc_tx_checksum(net_pkt_iface(pkt), is_ipv6 ? NET_IF_CHECKSUM_PROTOCOL_IPV6_TCP : NET_IF_CHECKSUM_PROTOCOL_IPV4_TCP) || force_chksum) {
```

Comment:
As mentioned above, IMO there's no need to overcomplicate with separate IPv4/IPv6 configs for TCP/UDP.